### PR TITLE
Release 4.2.0

### PR DESCRIPTION
**Pull request by wadells**
Following the instructions at https://wiki.gravitational.io/engineering/#release-checklist-teleport, these are all the source code changes that should be necessary to release 4.2.0.

Testing done:

Eyeballed the changes here:

https://github.com/gravitational/teleport/blob/wadells/cut-4.2.0/CHANGELOG.md

### CODE REVIEW

File: CHANGELOG.md
Comment (wadells) [inline CHANGELOG.md]: These links should be shortened for consistency with all the other links in the release notes.  We don't seem to use any raw URLs elsewhere.

File: CHANGELOG.md
Comment (wadells) [inline CHANGELOG.md]: Is this sufficiently documented?  I see reference to `--auth-server` and `--identity` for `tctl auth sign` and `tctl status` but there is no reference of these flags for other `tctl` commands or in the global flags?

File: CHANGELOG.md
Comment (webvictim) [inline CHANGELOG.md]: let's -> lets

File: CHANGELOG.md
Comment (webvictim) [inline CHANGELOG.md]: `Workflows API lets admins escalate RBAC roles in response to user requests.`

File: CHANGELOG.md
Comment (webvictim) [inline CHANGELOG.md]: Firestone -> Firestore

File: CHANGELOG.md
Comment (wadells) [reply CHANGELOG.md]: Addressed via a rebase & squash. 
3976997  has better links.

Comment (webvictim) [reply CHANGELOG.md]: I think anchoring with an absolute link to the specific docs version as we're doing here is fine.

File: CHANGELOG.md
Comment (webvictim) [reply CHANGELOG.md]: Yeah - I suspect there are a lot more instances where these flags could be used. We can push this for now though and update the docs separately later.

File: CHANGELOG.md
Comment (wadells) [inline CHANGELOG.md]: Shall we include GH ticket #s on these?  I can probably chase them down with @benarent's help.

File: CHANGELOG.md
Comment (benarent) [reply CHANGELOG.md]: Yes, this was a last min fix by me. I'll add a section into the admin manual for reasons why we've built this feature and How to use it.

File: CHANGELOG.md
Comment (webvictim) [reply CHANGELOG.md]: Yes, absolutely.

Comment (benarent) [reply CHANGELOG.md]: The issue numbers are here https://github.com/gravitational/teleport/issues/3133

File: CHANGELOG.md
Comment (benarent) [inline CHANGELOG.md]: Should we standardise our language to match SemVar https://semver.org/  in this case, this will be a Minor Teleport Release.

Comment (wadells) [reply CHANGELOG.md]: Done. f62c4620dd565a3894a900ab435f11a991e5e9e6 uses minor.

File: CHANGELOG.md
Comment (wadells) [reply CHANGELOG.md]: Corrected in f62c4620dd565a3894a900ab435f11a991e5e9e6

File: CHANGELOG.md
Comment (wadells) [reply CHANGELOG.md]: Corrected in f62c4620dd565a3894a900ab435f11a991e5e9e6

File: CHANGELOG.md
Comment (wadells) [reply CHANGELOG.md]: Fixed in f62c4620dd565a3894a900ab435f11a991e5e9e6

File: CHANGELOG.md
Comment (wadells) [reply CHANGELOG.md]: Included issue links for all 3 of these in f62c4620dd565a3894a900ab435f11a991e5e9e6

**Comment by benarent** [comment]
OK to test

**Comment by wadells** [comment]
@benarent @fspmarshall @webvictim: I'm waiting on at least one of you to approve this before I merge and proceed.

**Review by webvictim — APPROVED** [review]
LGTM

**Pull request merged**
